out/notebooks/spl2/installer.ts

```diff
@@ -0,0 +1,506 @@
+    const userSelection = (await popup) || null;
+    switch(userSelection) {
+        case agreeAndContinueChoice:
+            // Record preference so user is not asked again
+            workspace.getConfiguration().update(configKeyAcceptedTerms, TermsAcceptanceStatus.Accepted, true);
+            return true;
+        case viewTermsChoice:
+            console.log('Viewing Splunk General Terms in browser...');
+            env.openExternal(Uri.parse('https://www.splunk.com/en_us/legal/splunk-general-terms.html'));
+            return await promptToDownloadLsp(alsoInstallJava);
+        case turnOffSPL2Choice:
+            console.log('User opted out of SPL2 Langauge Server download, SPL2 support disabled');
+            // Record preference so user is not asked again
+            workspace.getConfiguration().update(configKeyAcceptedTerms, TermsAcceptanceStatus.DeclinedForever, true);
+            return false;
+        default:
+            // Cancel
+            return false;
+    }
+}
+
+/**
+ * Checks if the installed SPL2 Language Server version is the latest and prompt for
+ * upgrade if not, or automatically upgrade if user has that setting enabled.
+ */
+export async function getLatestSpl2Release(context: ExtensionContext, progressBar: StatusBarItem): Promise<void> {
+    return new Promise(async (resolve, reject) => {
+        const lspArtifactPath = path.join(context.globalStorageUri.fsPath, 'lsp');
+        // TODO: Remove this hardcoded version/update time and check for updates
+        let lspVersion: string = '2.0.362'; // context.globalState.get(stateKeyLatestLspVersion) || "";
+        const lastUpdateMs: number = Date.now(); // context.globalState.get(stateKeyLastLspCheck) || 0;
+        // Check for new version of SPL2 Language Server if longer than 24 hours
+        if (Date.now() - lastUpdateMs > 24 * 60 * 60 * 1000) {
+            const metaPath = path.join(lspArtifactPath, 'maven-metadata.xml');
+            try {
+                await downloadWithProgress(
+                    'https://splunk.jfrog.io/splunk/maven-splunk-release/spl2/com/splunk/spl/spl-lang-server-sockets/maven-metadata.xml',
+                    metaPath,
+                    progressBar,
+                    'Checking for SPL2 updates',
+                );
+                const parser = new XMLParser();
+                const metadata = fs.readFileSync(metaPath);
+                const metaParsed = parser.parse(metadata);
+                lspVersion = metaParsed?.metadata?.versioning?.release;
+            } catch (err) {
+                console.warn(`Error retrieving latest SPL2 version, err: ${err}`);
+            }
+        }
+        const lspFilename = `spl-lang-server-sockets-${lspVersion}-all.jar`;
+        const localLspPath = path.join(getLocalLspDir(context), lspFilename);
+        // Check if local file exists before downloading
+        if (!fs.existsSync(localLspPath)) {
+            try {
+                await downloadWithProgress(
```

Comment:
Maybe we should also delete older versions.

Reply:
ohh sorry, I think we delete at the to.

Reply:
that's right the entire lsp directory with any older versions is removed here: https://github.com/splunk/vscode-extension-splunk/pull/86/files/e055b64a0dc9afdccfee92f48aea47ff8b90928f#diff-46f9d5c9ce558798377cece1d98e6d5523c31f4f614a82164c897a1d351fcbe5R94